Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp

```diff
@@ -44,23 +44,17 @@ Structure* JSWebAssemblyStruct::createStructure(VM& vm, JSGlobalObject* globalOb
     return Structure::create(vm, globalObject, prototype, TypeInfo(WebAssemblyGCObjectType, StructureFlags), info());
 }
 
-JSWebAssemblyStruct::JSWebAssemblyStruct(VM& vm, Structure* structure, Ref<const Wasm::TypeDefinition>&& type, RefPtr<const Wasm::RTT> rtt)
-    : Base(vm, structure, rtt)
+JSWebAssemblyStruct::JSWebAssemblyStruct(VM& vm, Structure* structure, Ref<const Wasm::TypeDefinition>&& type, RefPtr<const Wasm::RTT>&& rtt)
+    : Base(vm, structure, WTFMove(rtt))
     , m_type(WTFMove(type))
     , m_payload(structType()->instancePayloadSize(), 0)
 {
 }
 
-JSWebAssemblyStruct* JSWebAssemblyStruct::tryCreate(JSGlobalObject* globalObject, Structure* structure, JSWebAssemblyInstance* instance, uint32_t typeIndex, RefPtr<const Wasm::RTT> rtt)
+JSWebAssemblyStruct* JSWebAssemblyStruct::create(VM& vm, Structure* structure, JSWebAssemblyInstance* instance, uint32_t typeIndex, RefPtr<const Wasm::RTT>&& rtt)
 {
-    VM& vm = globalObject->vm();
-
-    Ref<const Wasm::TypeDefinition> type = instance->module().moduleInformation().typeSignatures[typeIndex]->expand();
-
-    void* buffer = tryAllocateCell<JSWebAssemblyStruct>(vm);
-    if (UNLIKELY(!buffer))
-        return nullptr;
-    auto* structValue = new (NotNull, buffer) JSWebAssemblyStruct(vm, structure, Ref { type }, rtt);
+    Ref type = instance->module().moduleInformation().typeSignatures[typeIndex]->expand();
```

Comment:
This is a nice idiom to let the compiler infer the specialization of Ref.  It makes it explicit that we want a Ref here (which is the important detail), but without the clutter of the specialization details.  I should use this in my code too.